File: examples/java-spring/src/main/java/com/example/demo/domain/user/UserRepository.java
package com.example.demo.domain.user;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import java.util.Optional;

/**
 * Repository Interface
 * Part of Domain Layer - defines contract for persistence
 */
@Repository
public interface UserRepository extends JpaRepository<User, UserId> {
    Optional<User> findByEmail(Email email);
    boolean existsByEmail(Email email);
}

File: examples/nodejs-api/src/domain/user/UserRepository.ts
// Repository Interface (Domain Layer)
import { User } from './User';
import { UserId } from './UserId';
import { Email } from './Email';

export interface UserRepository {
  findById(id: UserId): Promise<User | null>;
  findByEmail(email: Email): Promise<User | null>;
  save(user: User): Promise<void>;
  delete(id: UserId): Promise<void>;
}
